ingester/ingester.go

```diff
@@ -557,8 +549,15 @@ func (i *Ingester) flushLoop(j int) {
 		}
 
 		userState.fpLocker.Lock(op.fp)
+
+		// Assume we're going to flush everything
 		chunks := series.chunkDescs
-		if !series.headChunkClosed {
+
+		// If the head chunk is old enough, close it
+		if op.immediate || model.Now().Sub(series.head().FirstTime()) > i.cfg.MaxChunkAge {
+			series.headChunkClosed = true
+			series.head().MaybePopulateLastTime()
```

Comment:
Kind of surprised `series` doesn't have a method that does both of these together, since it seems like a vaguely atomic operation.